tests/x11/multi_users_dm.pm

```diff
@@ -79,7 +80,9 @@ sub run {
     # Make sure screen changed before calling handle_login function (for slow workers)
     wait_still_screen;
     handle_login($user, 1);
-    assert_screen 'generic-desktop', 60;
+    my @desktop_needles = qw(generic-desktop);
+    push(@desktop_needles, 'opensuse-welcome') if opensuse_welcome_applicable;
+    assert_screen \@desktop_needles, 60;
```

Comment:
Maybe I should be more explicit. Just accepting the popup without closing it, will cause that the popup still appear the next time any module will switch to that user, right? If not, you will have those two lines added each time someone encounters the popup. But, the clean way to do it would be to close the popup and don't show on next startup.

Reply:
There are two different cases:
1. testing if login as a separate user works
2. running tests as a separate user

For (1.), the account is discarded immediately after. There is no reason to close the dialog, but there are at least two reasons **not closing** it: closing the dialog takes extra time, and trying to close it may fail (because input is unreliable). Closing the dialog is irrelevant, the module checks if the new user can login, and it should not do anything else. **Tests should be minimal.**

For (2.), the argument acually does not apply for several reason:
- is is purely hypothetic
- **iff** any tests were to be run, the module had to end after the login. As you said, you can not call a module. But this does not apply to the opensuse_welcome module alone, but to any other test/module you want to run.

Reply:
> Closing the dialog is irrelevant [...] Tests should be minimal
This is not a unit test. It is an acceptance test. Only weird users would leave the popup open while doing things on the computer.
Minimal code phylosophy != lazy coverage

> closing the dialog takes extra time, and trying to close it may fail (because input is unreliable).
- https://progress.opensuse.org/issues/56963#note-6

> is is purely hypothetic
- https://openqa.opensuse.org/tests/1033323#step/multi_users_dm/19
- https://openqa.opensuse.org/tests/1009439#step/first_boot/1
- https://openqa.opensuse.org/tests/1034499#step/opensuse_welcome/2
- https://openqa.opensuse.org/tests/1034417#step/installation/20


I could make an exception for change_password module (btw it is doing more than just changing password and should be split or renamed, but, at least is not doing anything else with user "test"), but not for multi_users_dm. It runs commands after that. The popup should not be in the background in an acceptance test.

Reply:
> > Closing the dialog is irrelevant [...] Tests should be minimal
> 
> This is not a unit test. It is an acceptance test. Only weird users would leave the popup open while doing things on the computer.
> Minimal code phylosophy != lazy coverage

By running tests multiple times, you don't increase coverage. You are just making the test unspecific.

Reply:
> > is is purely hypothetic
> 
>     * https://openqa.opensuse.org/tests/1033323#step/multi_users_dm/19

The test should be split up - login, opensuse_welcome, xterm

>     * https://openqa.opensuse.org/tests/1009439#step/first_boot/1

Why don't you link a current run, **which has succeeded as expected**?
https://openqa.opensuse.org/tests/1034316#step/opensuse_welcome/5
 
>     * https://openqa.opensuse.org/tests/1034499#step/opensuse_welcome/2

Different case. root -> skip the test.
 
>     * https://openqa.opensuse.org/tests/1034417#step/installation/20

https://openqa.opensuse.org/tests/1027306#step/installation/18

Accept 'opensuse-welcome' as "login succeded", and schedule opensuse_welcome afterwards.

Reply:
> By running tests multiple times, you don't increase coverage. You are just making the test unspecific.

It is not running a test several times. Closing the popup is part of what a use would do before continue working.

Reply:
So, as you obviously know much better what to do, I will leave ...

Reply:
> The test should be split up - login, opensuse_welcome, xterm

Having three modules each time you login as a user to to stuff is not minimal. And it will just contribute to the other pain we have with schedules. Do you know why the YAML schedule was invented?


> > * https://openqa.opensuse.org/tests/1009439#step/first_boot/1
> Why don't you link a current run, **which has succeeded as expected**?
> https://openqa.opensuse.org/tests/1034316#step/opensuse_welcome/5

The failure was not the point of this link. I don't agree with first_boot accepting the popup and then needing another module to close the popup. The module first_boot should close it.


> > * https://openqa.opensuse.org/tests/1034499#step/opensuse_welcome/2
> 
> Different case. root -> skip the test.

Exactly, and you are scheduling a module to an scenario that doesn't need it. Can you imagine if we schedule any special handling to any scenario just to skip when it is not necessary -> again the reboot nightmare.


> > * https://openqa.opensuse.org/tests/1034417#step/installation/20
> https://openqa.opensuse.org/tests/1027306#step/installation/18
> 
> Accept 'opensuse-welcome' as "login succeded", and schedule opensuse_welcome afterwards.

You are then forcing a team to change their test after your convenience. Or they will duplicate the handling to not follow your approach. This code is shared by several teams that will need to use and maintain the code.


Anyway. It is clear that we are not going anywhere. I cannot merge it knowing that it will cause future headaches, but someone else could. I am not the decision maker.
I was trying to make you know the consequences of doing this approach.